### PR TITLE
decorating EXTERNAL_MOVEMENTS_RECORD_INSERTED Event to reduce api cal…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…ls on the consumer side.

Example event:
{
        "eventType": "EXTERNAL_MOVEMENT_RECORD-INSERTED",
        "eventDatetime": "2020-02-25T16:02:52.818486",
        "offenderIdDisplay": "A5122DY",
        "bookingId": 1200869,
        "movementSeq": 1,
        "movementDateTime": "2020-02-25T15:57:45",
        "movementType": "ADM",
        "directionCode": "IN",
        "fromAgencyLocationId": "ALTRCT",
        "toAgencyLocationId": "MDI",
        "nomisEventType": "M1_RESULT"
    }

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import uk.gov.justice.digital.nomis.jpa.entity.OffenderExternalMovement;
+
+import java.util.Optional;
+
+@Repository
+public interface OffenderExternalMovementsRepository extends JpaRepository<OffenderExternalMovement, Long> {
+
+    Optional<OffenderExternalMovement> findByOffenderBookIdAndMovementSeq(Long movementSeq, Long bookingId);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java]: are those two fields indexed in the table?

File: src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.nomis.jpa.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import uk.gov.justice.digital.nomis.jpa.entity.OffenderExternalMovement;
+
+import java.util.Optional;
+
+@Repository
+public interface OffenderExternalMovementsRepository extends JpaRepository<OffenderExternalMovement, Long> {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java]: would be good to have a repository test for this change

Comment (lbennett-moj) [reply src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java]: There are no repository tests at all in this project.  I questioned this and after discussion thought the simplicity of this change didn't warrant the amount of work required.

File: src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java (hunk 1 above)
Comment (lbennett-moj) [reply src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java]: yes - there is a unique index

File: src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java (hunk 2 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/digital/nomis/jpa/repository/OffenderExternalMovementsRepository.java]: Mike Jackson didn't believe in repo tests.  He thought you were just testing the framework

File: src/main/java/uk/gov/justice/digital/nomis/service/XtagEventsService.java [hunk 3]
```diff
@@ -61,6 +61,17 @@ private OffenderEvent addAdditionalEventData(final OffenderEvent oe) {
                     .orElse(null);
             oe.setOffenderIdDisplay(nomsId);
         }
+        else if ("EXTERNAL_MOVEMENT_RECORD-INSERTED".equals(oe.getEventType())){
+            offenderService.getExternalMovement(oe.getBookingId(), oe.getMovementSeq()).ifPresent(em -> {
+                final var offenderBooking = em.getId().getOffenderBooking();
+                oe.setOffenderIdDisplay(offenderBooking.getOffender().getOffenderIdDisplay());
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/nomis/service/XtagEventsService.java]: How many SQL are generated for this.?

Comment (lbennett-moj) [reply src/main/java/uk/gov/justice/digital/nomis/service/XtagEventsService.java]: 2 statements. Unfortunately, unless I get the nomsId the rest of it is no help form Pathfinders point of view

**Pull request merged**